JavaImportTool/src/javaImportTool/ASTConverter.java

```diff
@@ -843,13 +854,8 @@ else if (e instanceof ArrayAccess)
 			node = new Node(null, "NewExpression", name);
 			
 			// Count how many dimensions are found in the the type itself
-			int typeDimensions = 1;
-			Type innerType = ac.getType().getComponentType();
-			while(innerType instanceof ArrayType)
-			{
-				++typeDimensions;
-				innerType = ((ArrayType) innerType).getComponentType();
-			}
+			int typeDimensions = ac.getType().getDimensions();
+			Type innerType = ac.getType().getElementType();
```

Comment:
As far as I understand `getDimensions()` is enough. This is not related to Java 8 but to an API change in the eclipse APIs (see [ArrayType](http://help.eclipse.org/mars/index.jsp?topic=%2Forg.eclipse.jdt.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fjdt%2Fcore%2Fdom%2FVariableDeclarationFragment.html))